Horatio/Horatio/Classes/Services/ServiceRequestOperation.swift

```
@@ -57,7 +57,7 @@ open class FetchServiceResponseOperation: GroupOperation {
     // MARK: - Private
 
     fileprivate static func generatedCacheFileName(_ request: ServiceRequest) -> String {
-        guard let url = request.url else { return UUID().uuidString }
+        guard let url = request.url, !(url.hasDirectoryPath && url.lastPathComponent.count == 1) else { return UUID().uuidString }
```

Prefer to pull this into a let above the guard; for example,

```swift
let hasCachableComponent = url.hasDirectoryPath && url.lastPathComponent.count == 1
guard let url = request.url, hasCacheableComponent else { return ... }
```

---

Yeah I tried that, but `url` is bound in the guard. I could do two guards to break it up, but that means two identical return statements.

---

So it's either...
```swift
    fileprivate static func generatedCacheFileName(_ request: ServiceRequest) -> String {
        if let url = request.url {
            let cacheComponent = url.lastPathComponent
            
            // Condition indicates that cacheComponent is "/", which is invalid
            let hasValidCacheComponent = !(url.hasDirectoryPath && cacheComponent.count == 1)
            
            if hasValidCacheComponent {
                return cacheComponent
            }
        }

        return UUID().uuidString
    }
```
Or...
```swift
    fileprivate static func generatedCacheFileName(_ request: ServiceRequest) -> String {
        guard let url = request.url else { return UUID().uuidString }
        
        let cacheComponent = url.lastPathComponent
        let hasValidCacheComponent = !(url.hasDirectoryPath && cacheComponent.count == 1)
        guard hasValidCacheComponent else { return UUID().uuidString }
        
        return cacheComponent
    }
```

So the question is fail early, or failsafe?

---

Yeah, much preferred. I don't like inverting bools like that on setting (prefer to `!foo` in the test instead), but I can't think of a good name that represents that, so I'll let it slide... :)

---

I like the second one better (the first one implies that the "normal return" is the UUID)*

*(I can't remember what Code Complete calls it, but the last return in the method is typically the "if nothing goes wrong, you expect this one")